src/bsp/src/stm32/hal/src/hal_flash.c

```diff
@@ -21,14 +21,14 @@ bool Flash_program(uint32_t address, uint8_t* data, uint32_t size) {
 
     HAL_FLASH_Unlock();
 
-    for (unsigned int i = 0; i < size; i++) {
-        if (HAL_FLASH_Program(FLASH_TYPEPROGRAM_BYTE, address, (uint64_t)(*data)) != HAL_OK) {
+    for (unsigned int i = 0; i < size; i+=4) {
+        if (HAL_FLASH_Program(FLASH_PROGRAM_32_BYTES, address, (uint64_t)(*data)) != HAL_OK) {
             HAL_FLASH_Lock();
             return false;
         }
 
-        address++;
-        data++;
+        address+=4;
+        data+=4;
```

Comment:
@cquesnel Fix since we can't program per byte, tagging you so you can double check it.

Reply:
I don't think it will cause a segfault unless the data we are writing is at the limit of the RAM. If it ever does, we could just pad the structure we are writing so it is a multiple of 4